app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,55 @@
 class PostsController < ApplicationController
   def index
-    @posts = Post.where(author_id: params[:user_id])
+    @posts = Post.includes(:author).where(author_id: params[:user_id])
+    @user = User.includes(:posts).find(params[:user_id])
   end
 
   def show
-    @post = Post.find(params[:id])
+    @post = Post.includes(:author).find_by(author_id: params[:user_id], id: params[:id])
+
+    if @post
+      @user = @post.author
+      @comments = @post.comments
+    else
+      flash[:alert] = 'Post not found'
+      redirect_to user_posts_path(params[:user_id])
+    end
+  end
+
+  def new
+    @user = current_user
+    @post = Post.new
+  end
+
+  def create
+    @user = current_user
+    @post = @user.posts.new(post_params)
+    if @post.save
+      redirect_to user_post_path(@user, @post)
+    else
+      puts @user
+      puts @post.errors.full_messages
+      flash.now[:errors] = 'Invalid post!'
+      render :new
+    end
+  end
```

Comment:
**OPTIONAL** - Good job done so far, but when creating a new post I get an error, in the screenshot attached below, this is because you do not set attributes for the likes and comments counter when a post is created, please fix this 🙏 

![image](https://github.com/firstpj/blog-app/assets/73607512/0edcd86e-1a5c-4131-8b7b-c0fe3b837aaf)